src/main/kotlin/com/atlassian/performance/tools/virtualusers/api/config/VirtualUserBehavior.kt

```diff
@@ -105,8 +105,10 @@ class VirtualUserBehavior private constructor(
         load: VirtualUserLoad
     ): VirtualUserBehavior = Builder(this).load(load).build()
 
+    @Deprecated("TODO - can we provide more type safe builder?")
+    // TODO detect when the new scenario is used with parameters that are no longer supported in new Scenario
```

Comment:
IMO we should keep the old builder and introduce a new one, without the non-supported parameters.